src/firefly/js/visualize/DrawLayerCntlr.js

```diff
@@ -316,33 +342,64 @@ export function dispatchDeleteRegionLayer(drawLayerId, plotId, dispatcher = flux
+ * @param {string} layerTitle    will replace the original title if the drawing layer exists and layerTitle is non-empty.
+ * @param {Object} selectMode    redndering features for the selected region
+ * @param {string} selectMode.selectStyle   rendering style for the selected region including 'UprightBox', 'DottedOverlay',
+ * 'SolidOverlay', 'DottedReplace', and 'SolidReplace'
+ * @param {string} selectMode.selectColor   rendering color for the selected region, ex: 'red', '#DAA520'
+ * @param {int}    selectMode.lineWidth     rendering line width for the selected region. 0 or less means the line width
+ * is the same as that of the selected region
+ * @param {function} dispatcher
+ * @public
  * @func dispatchAddRegionEntry
  * @memberof firefly.action
  */
-export function dispatchAddRegionEntry(drawLayerId, regionChanges, plotId=[], layerTitle='', dispatcher = flux.process) {
-    dispatcher({type: REGION_ADD_ENTRY, payload: {drawLayerId, regionChanges, plotId, layerTitle}});
+export function dispatchAddRegionEntry(drawLayerId, regionChanges, plotId=[], layerTitle='',
+                                        {selectStyle = defaultRegionSelectStyle, selectColor = defaultRegionSelectColor,
+                                        lineWidth = 0 },     // lineWidth <= 0, as original region
+                                       dispatcher = flux.process) {
+    dispatcher({type: REGION_ADD_ENTRY, payload: {drawLayerId, regionChanges, plotId, layerTitle,
+                                                  selectMode: {selectStyle: getRegionSelectStyle(selectStyle),
+                                                               selectColor, lineWidth}}});
 }
 
 /**
- * remove the region entry from the plot layer with drawLayerId
- * @param drawLayerId
- * @param regionChanges
- * @param dispatcher
+ * @summary remove region(s) from the drawing layer
+ * @param {string} drawLayerId id of the drawing layer where the region(s) are removed from, required.
+ * @param {string[]|string} regionChanges array or string of region description.
+ * @param {function} dispatcher
+ * @public
  * @func dispatchRemoveRegionEntry
  * @memberof firefly.action
  */
 export function dispatchRemoveRegionEntry(drawLayerId, regionChanges, dispatcher = flux.process) {
     dispatcher({type: REGION_REMOVE_ENTRY, payload: {drawLayerId, regionChanges}});
 }
+
+
+/**
+ * @summary select region from a drawing layer containing regions
+ * @param {string} drawLayerId  id of drawing layer where the region is selected from, required.
+ * @param {string[]|string} selectedRegion array or string of region description, currently only single region is allowed
+ * to be selected if the array contains the description of multiple regions.
+ * @param {function} dispatcher
+ * @public
+ * @func dispatchSelectRegion
+ * @memberof firefly.action
+ */
+export function dispatchSelectRegion(drawLayerId, selectedRegion, dispatcher = flux.process) {
```

Comment:
- The document is missing parameter type for selectedRegion. selectedRegion could be a string, array of strings, or a region object. I will fix it. 
- when clicking on a region, the action payload will contain the region object. In order to know which region is selected, there is a function reporting the status of the selected region, it is 'getSelectedRegion' from RegionPlot.js which returns the region string, that is the string parsed to build the region object. The string returned from 'getSelectedRegion'  is that either formed (normalized) from a region file on the server  or  from array of strings in the front side. Each returned string has  the coordinate info. (image, physical, j2000, ...) attached  in the front.